internal/pkg/manifest/lb_web_svc.go

```diff
@@ -46,13 +61,60 @@ func (lc *LoadBalancedWebServiceConfig) LogConfigOpts() *template.LogConfigOpts
 	return lc.logConfigOpts()
 }
 
+func (lc *LoadBalancedWebServiceConfig) HTTPHealthCheckOpts() *template.HTTPHealthCheckOpts {
+	opts := template.HTTPHealthCheckOpts{
+		HealthCheckPath:    aws.String(defaultHealthCheckPath),
+		HealthyThreshold:   aws.Int64(defaultHealthyThreshold),
+		UnhealthyThreshold: aws.Int64(defaultUnhealthyThreshold),
+		Interval:           aws.Int64(defaultInterval),
+		Timeout:            aws.Int64(defaultTimeout),
+	}
+	if lc.RoutingRule.HealthCheck.HealthCheckArgs.Path != nil {
+		opts.HealthCheckPath = lc.RoutingRule.HealthCheck.HealthCheckArgs.Path
+	}
+	if lc.RoutingRule.HealthCheck.HealthCheckPath != nil {
+		opts.HealthCheckPath = lc.HealthCheck.HealthCheckPath
+	}
+	if lc.RoutingRule.HealthCheck.HealthCheckArgs.HealthyThreshold != nil {
+		opts.HealthyThreshold = lc.RoutingRule.HealthCheck.HealthCheckArgs.HealthyThreshold
+	}
+	if lc.RoutingRule.HealthCheck.HealthCheckArgs.UnhealthyThreshold != nil {
+		opts.UnhealthyThreshold = lc.RoutingRule.HealthCheck.HealthCheckArgs.UnhealthyThreshold
+	}
+	if lc.RoutingRule.HealthCheck.HealthCheckArgs.Interval != nil {
+		opts.Interval = aws.Int64(int64(lc.RoutingRule.HealthCheck.HealthCheckArgs.Interval.Seconds()))
+	}
+	if lc.RoutingRule.HealthCheck.HealthCheckArgs.Timeout != nil {
+		opts.Timeout = aws.Int64(int64(lc.RoutingRule.HealthCheck.HealthCheckArgs.Timeout.Seconds()))
+	}
+	return &opts
+}
+
+// HTTPHealthCheckArgs holds the configuration to determine if the load balanced web service is healthy.
+// These options are specifiable under the "healthcheck" field.
+// See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticloadbalancingv2-targetgroup.html.
+type HTTPHealthCheckArgs struct {
+	Path               *string        `yaml:"path"`
+	HealthyThreshold   *int64         `yaml:"healthy_threshold"`
+	UnhealthyThreshold *int64         `yaml:"unhealthy_threshold"`
+	Timeout            *time.Duration `yaml:"timeout"`
+	Interval           *time.Duration `yaml:"interval"`
+}
+
+// HealthCheckArgsOrString is a custom type which supports unmarshaling yaml which
+// can either be of type string or type HealthCheckArgs. q
+type HealthCheckArgsOrString struct {
+	HealthCheckPath *string
+	HealthCheckArgs HTTPHealthCheckArgs
+}
+
 // RoutingRule holds the path to route requests to the service.
 type RoutingRule struct {
-	Path            *string `yaml:"path"`
-	HealthCheckPath *string `yaml:"healthcheck"`
-	Stickiness      *bool   `yaml:"stickiness"`
+	Path        *string                 `yaml:"path"`
+	HealthCheck HealthCheckArgsOrString `yaml:"healthcheck"`
+	Stickiness  *bool                   `yaml:"stickiness"`
 	// TargetContainer is the container load balancer routes traffic to.
-	TargetContainer *string `yaml:"targetContainer"`
+	TargetContainer *string `yaml:"target_container"`
```

Comment:
Ah I wish we could change this :'( but we would need to support both `targetContainer` and `target_container` to maintain backwards compatibility. 

Right now this will break users that have `targetContainer` as a field. We'd need to figure out a way of assigning two yaml fields to `TargetContainer`.
Maybe this works?
```suggestion
	TargetContainer *string `yaml:"target_container" yaml:"targetContainer"` // "targetContainer" for backwards compatibility
```
If it does great, otherwise can we punt this change to a separate PR?

Reply:
Okay, so that :point_up: didn't work and some digging/reading makes me think there isn't a built-in way to do this, but this little hack works:
In `manifest/lb_web_svc.go`, I added another field to the struct for camel case.
In `stack/lb_web_svc.go`:
```
if s.manifest.TargetContainer == nil && s.manifest.TargetContainerCamelCase != nil {
		s.manifest.TargetContainer = s.manifest.TargetContainerCamelCase
	}
```

Reply:
brilliant! that sounds good to me!

Reply:
samesies thanks for diving deep on this!